js/fibonacci.js

```diff
@@ -0,0 +1,178 @@
+    elementOfResult.classList.remove("text-danger");
+    elementOfResult.innerHTML = "";
+    if (elementOfIndex.value > 50) {
+        elementOfIndex.classList.add("is-invalid");
+        elementOfIndex.classList.add("text-danger");
+        validFeedback.innerText = "Can't be larger than 50";
+        loadingSpinner.classList.add("visually-hidden");
+    } else if (elementOfIndex.value < 0) {
+        elementOfIndex.classList.add("is-invalid");
+        validFeedback.innerText = "Can't be a negative value";
+        loadingSpinner.classList.add("visually-hidden");
+    } else if (elementOfIndex.value === "") {
+        elementOfIndex.classList.add("is-invalid");
+        validFeedback.innerText = "Can't be empty";
+        loadingSpinner.classList.add("visually-hidden");
+    } else {
+        elementOfResult.classList.remove("fs-5");
+        if (saveCalc.checked) {
+            getFibonacciFromServer(elementOfIndex.value)
+                .then(result => {
+                    elementOfResult.innerHTML = `&nbsp;<u><b>${result}</b></u>`;
+                    loadingSpinner.classList.add("visually-hidden");
+                    if (parseInt(elementOfIndex.value) === 42) {
+                        elementOfResult.classList.add("text-danger");
+                        elementOfResult.classList.add("fs-5");
+                    } else {
+                        getFibonacciResults().then(arrayOfResponses => sortResults(arrayOfResponses)).then(arrayOfResponses => addFibonacciResult(arrayOfResponses));
+                    }
+                });
+        } else {
+            elementOfResult.innerHTML = `&nbsp;<u><b>${getFibonacci(elementOfIndex.value)}</b></u>`;
+            loadingSpinner.classList.add("visually-hidden");
+        }
+    }
+};
+
+// =================================================================
+
+//Sort menu
+sortSelectElement.addEventListener("change", sortRes);
+function sortRes() {
+    calcRes.innerHTML = "";
+    getFibonacciResults()
+        .then(arrayOfResponses => sortResults(arrayOfResponses))
+        .then(arrayOfResponses => displayResults(arrayOfResponses));
+};
+
+function sortResults(arrayToSort) {
+    const myOption = sortSelectElement.options[sortSelectElement.selectedIndex];
+    switch (myOption.value) {
+        case "Number Asc":
+            return arrayToSort.sort(function (firstObj, lastObj) { return firstObj.number - lastObj.number });
+        case "Number Desc":
+            return arrayToSort.sort(function (firstObj, lastObj) { return lastObj.number - firstObj.number });
+        case "Date Asc":
+            return arrayToSort.sort(function (firstObj, lastObj) { return firstObj.createdDate - lastObj.createdDate });
+        case "Date Desc":
+            return arrayToSort.sort(function (firstObj, lastObj) { return lastObj.createdDate - firstObj.createdDate });
+    }
+};
+
+// =================================================================
+async function getFibonacciFromServer(num) {
```

Comment:
inside a project. it's better to not mix between async await and .then() syntax. Choose one of the options and use it across all of your code.